### DISCUSSION TITLE
Unable to Download the CLI Validator Tool

### DISCUSSION THREAD

**Discussion by DigiFool23** [CMSgov/hospital-price-transparency#149] [Q&A] [answered]
I manage the public website for my organization, and my team needs to validate our pricing transparency files to ensure they are compliant, but our files seem to be too large to use the online validator (I know there is technically no file limit for this tool, but CMS notes that the CLI validator works best for large files), and our data engineers cannot download / install the CLI validator tool.

Here is their feedback on the issues they are experiencing when trying to install the CLI validator tool:

"[CMS'] instructions don’t match what happens when you follow them. It’s not like installing a typical program. They posted a bunch of source code."

This data engineer has worked with 2 other groups of data engineers, and none of them are able to install this program due to the above noted issue. 

We need to get these files validated ASAP. I have also submitted a support ticket to CMS, as well as followed-up with their below emails, but haven't heard back yet:

- PriceTransparencyHospitalCharges@cms.hhs.gov (just emailed this one today)
- PriceTransparencyInCoverage@CMS.hhs.gov (have followed-up twice in the last week, no response)

Can someone explain how they have successfully installed this tool? Any instructions that you could offer, tips, system requirements, etc would be very helpful, so that I can share this information with our data engineers.

**Comment by daniel-eckel** [comment, marked as answer]
The formal instructions for running the Command Line Interface (CLI) Validator are found on it’s [homepage](https://github.com/CMSgov/hpt-validator-cli/?tab=readme-ov-file#getting-started). The dependencies required to run the CLI Validator are supported by all major operating systems (e.g., Mac, Windows, Linux). See below for step-by-step instructions.

1.	**Install Prerequisites:** Download and install node.js from here: [Node.js — Download Node.js®.](https://nodejs.org/en/download/prebuilt-installer)
2.	**Open a Command Prompt Window**: Open any command line interface shell such as the ‘Node.js command prompt’ that was just installed.
3.	**Install CLI Validator:** Type `npm install -g hpt-validator-cli` then push enter. This will install the CLI Validator. 
4.	**Specify File Path (Optional):** The validator runs files from the location specified by the file path in the command prompt. In Windows the default for many command prompts is C:\USERS\ [USERNAME]. To choose your own file path, type `cd` followed by the file path. For example, for a user with the username “CMS”, enter `cd “C:\Users\CMS\Downloads”`. The validator will now look in the downloads folder of the CMS user for the file to test.
5.	**Test File:** Type `cms-hpt-validator`, the file name and `v2.0`, then enter, to run the file. For example, entering `cms-hpt-validator 123456789_west-mercy-hospital-_standardcharges.csv v2.0` will test the file named "123456789_west-mercy-hospital-_standardcharges.csv".
6.	**Limit Errors Shown (Optional):** By default, the validator displays every error found in the machine-readable file being tested. Users may limit the number of errors returned but adding `-e` and the desired number. For example, to limit the CLI validator to showing 50 errors and 50 warnings enter `cms-hpt-validator 123456789_west-mercy-hospital-_standardcharges.csv v2.0 -e 50`
